From 0dcdf2fd04218b53ea67ef98a63f4074dffc48e7 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Thu, 27 May 2021 19:06:56 +0300
Subject: [PATCH] [SW-43829] lkd: preserve boot_info in handler

Recently, we added preserving boot_info as part of
handling of wfe and reset, but erroneously it was added
only in lkd legacy protocol handler, and therefore
when using COMMS, boot_info is not preserved.
This patch fix it by calling the preserving inside the
actual handlers of wfe and reset.

Change-Id: Ifc352a256ebb74e9f15f2244af6b4c4b0fd6ca63
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 053716d..9031925 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -532,18 +532,6 @@
 	fit_run_image();
 }
 
-static void module_lkd_run_noret_reset(void)
-{
-	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
-	hl_dev_reset();	/* should not return */
-}
-
-static void module_lkd_run_noret_wfe(void)
-{
-	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
-	wfe();
-}
-
 __weak void preboot_finalize_nic_en(void) {}
 
 static void module_lkd_preserve_sp_info(void)
@@ -554,15 +542,27 @@
 	boot_info_finalize();
 }
 
+static void module_lkd_run_noret_reset(void)
+{
+	module_lkd_preserve_sp_info();
+	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
+	hl_dev_reset();	/* should not return */
+}
+
+static void module_lkd_run_noret_wfe(void)
+{
+	module_lkd_preserve_sp_info();
+	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
+	wfe();
+}
+
 static void module_lkd_run_noret_legacy(void)
 {
 	switch (sts.cmd.val) {
 	case KMD_MSG_GOTO_WFE:
-		module_lkd_preserve_sp_info();
 		module_lkd_run_noret_wfe();
 		break;
 	case KMD_MSG_RST_DEV:
-		module_lkd_preserve_sp_info();
 		module_lkd_run_noret_reset();
 		break;
 	case KMD_MSG_FIT_RDY:
